pkg/gardenlet/apis/config/v1alpha1/types.go

```diff
@@ -66,15 +66,8 @@ type GardenletConfiguration struct {
 	// Default: nil
 	// +optional
 	FeatureGates map[string]bool `json:"featureGates,omitempty"`
-	// SeedConfig contains configuration for the seed cluster. Must not be set if seed selector is set.
-	// In this case the gardenlet creates the `Seed` object itself based on the provided config.
-	// +optional
+	// SeedConfig contains configuration for the seed cluster.
```

Comment:
Is it mandatory now? Then we should remove the pointer and the `omitempty`, or?

Reply:
If we do that, how do we define the "no seed config given" situation for cases when users forgot to configure it? If `seedConfig.Name` is empty? Should I just remove all validation that checks that the `seedConfig` is set? :thinking:

Reply:
See https://github.com/gardener/gardener/pull/4306/commits/f2db36e6e11ad96a965e6768c85f13cd6c70881f#diff-8cb561a8767668327c5c030c4b54c0fbc2b3fd7283c28cce1e8e4ac4a6e2fec5R152 and https://github.com/gardener/gardener/pull/4306/commits/f2db36e6e11ad96a965e6768c85f13cd6c70881f#diff-dbdd0162253dcb4ba9b6ed195d49fba0b4cb2efcbdc529f31d83f4045101cf95R48 for concrete examples where I'm unsure how to proceed.

Reply:
`seedConfig` has to stay a pointer, because it is optional in the `ManagedSeed.spec.gardenlet.config` case (is filled by the ManagedSeed controller).

Reply:
OK, thanks for clarifying, I was just wondering about it because the `// +optional` tag was removed

Reply:
Ah, good point. The `// +optional` should probably be kept, as long as it's a pointer (and is actually optional in some circumstances), right?

Reply:
Already getting restored, I'm just fiddling with the seed_clientmap tests.